### PR TITLE
Isaacs/tree shaking pr 295 continued

### PR DESCRIPTION
**Pull request by isaacs**
Following on from #295, taking the tree shaking and test coverage a bit further.

Because why not?

Not yet ready to land.  Need test coverage for the classes.

### CODE REVIEW

**Comment by isaacs** [comment]
Alright, back up to 100% coverage, this time with a coverage map so we're targeting each component specifically with tests.

**Comment by isaacs** [comment]
Care to review?  @stabbylambda @npm/cli-team

**Comment by isaacs** [comment]
(Probably better to just review the tree rather than try to look at the diffs.  It's... a lot.)

**Comment by darcyclarke** [comment]
@isaacs mind adding a `.github/settings.yml` with the following reference(?):

```
---
_extends: 'open-source-project-boilerplate'
```

Would be good to get those default labels added...

**Comment by stabbylambda** [comment]
I love that because so many of the class functions are dupes of the standalone functions, we're now up to 5535 tests.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const lrCache = {}
+const lazyRequire = (path, subkey) => {
```
Comment (Aghassi) [inline index.js]: I just want to say I find this very slick and I like it

**Comment by isaacs** [comment]
> @isaacs mind adding a `.github/settings.yml` with the following reference(?):

Done!

**Comment by stabbylambda** [comment]
Just getting out of my post-Thanksgiving coma. Wondering if there's anything else that needs to be done here. 😄

**Comment by isaacs** [comment]
Shipped as v7.

**Comment by stabbylambda** [comment]
Awesome! Thanks for working with me on this! 👍

**Pull request closed without merging**